hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/MergeOnReadSnapshotRelation.scala

```diff
@@ -213,7 +214,24 @@ class MergeOnReadSnapshotRelation(sqlContext: SQLContext,
         PartitionedFile(getPartitionColumnsAsInternalRow(file.getFileStatus), filePath, 0, file.getFileLen)
```

Comment:
Let's also update `getPartitionColumnsAsInternalRow` to reuse `parsePartitionColumnValues` instead